pytorch_blade/src/compiler/jit/torch/shape_analysis_test.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 The BladeDISC Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <torch/csrc/jit/ir/irparser.h>
+#include <torch/csrc/jit/testing/file_check.h>
+#include <torch/script.h>
+#include "compiler/jit/torch/shape_analysis.h"
+
+namespace torch {
+namespace blade {
+
+TEST(PropagateInputShapesTest, SimpleUnary) {
+#if PYTORCH_MAJOR_VERSOIN == 1 && PYTORCH_MINOR_VERSION >= 8
+  const std::string graph_str = R"IR(
+graph(%p1 : Float(*, *, *, device=cuda:0)):
```

Comment:
Will it run successfully on cpu version pytorch?

Reply:
it works well, shape analysis pass does care about the device.